app/components/message_filter.py

```diff
@@ -1,76 +1,95 @@
+from __future__ import annotations
+
 import re
-from collections.abc import Callable
-from typing import NamedTuple, cast
+from typing import TYPE_CHECKING, NamedTuple, cast, final
 
 import discord as dc
+from discord.ext import commands
 
 from app.common.message_moving import format_or_file
-from app.setup import config
 from app.utils import try_dm
 
-URL_REGEX = re.compile(
+if TYPE_CHECKING:
+    from collections.abc import Callable
+
+    from app.bot import GhosttyBot
+
+_URL_REGEX = re.compile(
     r"https?:\/\/(?:www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b"
     r"(?:[-a-zA-Z0-9()@:%_\+.~#?&//=]*)"
 )
-MESSAGE_DELETION_TEMPLATE = (
+_MESSAGE_DELETION_TEMPLATE = (
     "Hey! Your message in {} was deleted because it did not contain {}. "
     "Make sure to include {}, and respond in threads.\n"
 )
-MESSAGE_CONTENT_NOTICE = "Here's the message you tried to send:"
-COPY_TEXT_HINT = (
+_MESSAGE_CONTENT_NOTICE = "Here's the message you tried to send:"
+_COPY_TEXT_HINT = (
     "-# **Hint:** you can get your original message with formatting preserved "
     'by using the "Copy Text" action in the context menu.'
 )
-REGULAR_MESSAGE_TYPES = frozenset({dc.MessageType.default, dc.MessageType.reply})
+_REGULAR_MESSAGE_TYPES = frozenset({dc.MessageType.default, dc.MessageType.reply})
 
 
-class MessageFilter(NamedTuple):
+class MessageFilterTuple(NamedTuple):
```

Comment:
In line with the constants above:

```suggestion
class _MessageFilterTuple(NamedTuple):
```

I personally think we can just call them `_Filter` too if it is private to a file about message filters.